c10/core/CPUAllocator.cpp

```diff
@@ -292,6 +292,10 @@ void ProfiledCPUMemoryReporter::Delete(void* ptr) {
       allocated = allocated_;
       nbytes = it->second;
       size_table_.erase(it);
+    } else {
```

Comment:
are there any changes required in CUDAAllocator?

Reply:
CUDACachingAllocator already saves block sizes